pkg/kube/fake/printer.go

```diff
@@ -30,7 +30,8 @@ import (
 // PrintingKubeClient implements KubeClient, but simply prints the reader to
 // the given output.
 type PrintingKubeClient struct {
-	Out io.Writer
+	Out          io.Writer
+	WaitDuration time.Duration
```

Comment:
Why was this introduced instead of using the time.Duration in the function signature for `Wait`? What's the expected behaviour of the following code:

```
p := PrintingKubeClient{ WaitDuration: 10*time.Second, Out: ioutil.Discard }
p.Wait(resources, 10*time.Second)
```

Reply:
This was introduced to fake a waiting period. In my case, it is to make the process of installing or upgrading wait, so the other part of the test sends the SIGTERM.

```
p := PrintingKubeClient{ WaitDuration: 10*time.Second, Out: ioutil.Discard }
p.Wait(resources, 10*time.Second)
```
The expected behavior is to wait 10 secs to received the response.

Reply:
After triple-checking the code, I decided to retract my changes in the PrintingKubeClient.
Instead the logic to simulate a delay in the wait for resources will be in the FakeKubeClient